### PR TITLE
Validator: Added warnings and info messages for Mutation_Status column for mutations_extended validation

### PR DESCRIPTION
**Pull request by dionnezaal**
# What? Why?
When loading mutation data we encountered that when mutation files contain 'LOH' or 'None' ([allowed values for Mutation_Status in MAF format](https://wiki.nci.nih.gov/display/TCGA/Mutation+Annotation+Format+(MAF)+Specification)) in the Mutation_Status where not loaded by the importer, but no warning was given to the user in the validation script.  Also, only Mutation_Status 'Somatic' and 'Germline' are used by cBioPortal and given a symbol in the mutation tab, but if values differ from these no warning is given to the user. 

## Changes proposed in this pull request
- Give warning to the user that line in mutation file will not be loaded when mutation file contains 'LOH' or 'None' in Mutation_Status column. 
- Give warning to the user when Mutation_Status column contains different values than 'Somatic' or 'Germline'.

screen shot 2018-01-09 at 12 30 09 pm

### CODE REVIEW

**Comment by dionnezaal** [comment]
After discussion with @sandertan changed the mutation not loaded to a info message. Also changed warning when Mutation_Status column contains different values than 'Somatic' or 'Germline' to allow  'Wildtype' and all allowed Mutation_Status values from the MAF format.

File: core/src/test/scripts/unit_tests_validate_data.py [hunk 1]
```diff
@@ -1349,6 +1349,55 @@ def test_validation_verification_status(self):
...
+        LOH, None or Wildtype. In these cases the mutation is not loaded in cBioPortal."""
+        self.logger.setLevel(logging.INFO)
+        record_list = self.validate('mutations/data_mutations_not_loaded_ms.maf',
+                                    validateData.MutationsExtendedValidator,
+                                    extra_meta_fields={'swissprot_identifier': 'name'})
+
+        # We expect 5 info messages, 3 from not loaded mutations and 2 general lines
+        self.assertEqual(len(record_list), 5)
+        record_iterator = iter(record_list)
+        # Validation Status error
```
Comment (sandertan) [inline core/src/test/scripts/unit_tests_validate_data.py]: Please change this comment and the 2 below to something like "# Expecting info message"

File: core/src/test/scripts/unit_tests_validate_data.py [hunk 2]
```diff
@@ -1349,6 +1349,55 @@ def test_validation_verification_status(self):
...
 
+    def test_validation_ms(self):
+
+        """Test if warning is given when Mutation_Status column in mutation
+        file contains other than allowed values from MAF format or Wildtype."""
+        self.logger.setLevel(logging.WARNING)
+        record_list = self.validate('mutations/data_mutations_ms.maf',
+                                    validateData.MutationsExtendedValidator,
+                                    extra_meta_fields={'swissprot_identifier': 'name'})
+        # we expect 1 error
```
Comment (sandertan) [inline core/src/test/scripts/unit_tests_validate_data.py]: warning

File: core/src/main/scripts/importer/validateData.py [hunk 3]
```diff
@@ -1507,6 +1508,16 @@ def checkFilterAnnotation(self, value):
             return False
         return True
 
+    def checkMutationStatus(self, value):
+        """Check values in mutation status column."""
+        if value in ['LOH', 'None', 'Wildtype']:
+            self.logger.info('Mutation will not be loaded due to Mutation Status',
```
Comment (sandertan) [inline core/src/main/scripts/importer/validateData.py]: Can we make it `Mutation will not be loaded due to value in Mutation_Status`?

File: core/src/main/scripts/importer/validateData.py [hunk 4]
```diff
@@ -1507,6 +1508,16 @@ def checkFilterAnnotation(self, value):
...
+                                extra={'line_number': self.line_number, 'cause': value})
+        if value not in ['None', 'Germline', 'Somatic', 'LOH', 'Post-transcriptional modification', 'Unknown', 'Wildtype'] and value != '':
+            self.logger.warning('Mutation status is not in MAF format',
```
Comment (sandertan) [inline core/src/main/scripts/importer/validateData.py]: `Mutation_Status value is not in MAF format`

File: core/src/test/scripts/unit_tests_validate_data.py [hunk 5]
```diff
@@ -1378,24 +1378,24 @@ def test_mutation_not_loaded_ms(self):
...
         record_iterator = iter(record_list)
-        # Validation Status error
+        # Expected info message due to value "None" in Mutation_Status
         record = record_iterator.next()
         self.assertEqual(record.levelno, logging.INFO)
         self.assertEqual(record.line_number, 3)
         self.assertEqual(record.cause, 'None')
-        self.assertEqual(record.getMessage(), "Mutation will not be loaded due to Mutation Status")
-        # Validation Status error
+        self.assertEqual(record.getMessage(), "Mutation will not be loaded due to value in Mutation Status")
```
Comment (sandertan) [inline core/src/test/scripts/unit_tests_validate_data.py]: Lets make this `Mutation_Status` for consistency.

**Comment by dionnezaal** [comment]
Applied last requested change from @sandertan, also added update for documentation about the `Mutation_Status` to File-Formats.md, squashed all commits and rebased on master.

**Comment by dionnezaal** [comment]
Rebased PR on master and solved merge conflict in docs/File-Formats.md

**Pull request merged**